MailChimp.Net/Configuration/MailChimpConfiguration.cs

```diff
@@ -1,17 +1,36 @@
 ﻿using System;
-using System.Configuration;
+using System.Linq;
 using MailChimp.Net.Interfaces;
+using Microsoft.Extensions.Configuration;
 
 namespace MailChimp.Net
 {
     public class MailChimpConfiguration : IMailChimpConfiguration
     {
-        public static int DefaultLimit => Common.DefaultLimit;
+        public MailChimpConfiguration()
+        {
+            var builder = new ConfigurationBuilder()
+                .AddJsonFile("appsettings.json", optional: true, reloadOnChange: true)
+                .AddJsonFile("logging.json", optional: true, reloadOnChange: true)
+                .AddEnvironmentVariables();
+
+
+
+            Configuration = builder.Build();
+        }
+
+        public IConfiguration Configuration { get; }
+
+        public static int DefaultLimit { get { return Common.DefaultLimit; } }
 
         private string _apiKey;
         public string ApiKey
         {
-            get { return _apiKey ?? (_apiKey = ConfigurationManager.AppSettings["MailChimpApiKey"]); }
+            get { return _apiKey ?? (_apiKey = Configuration
+                    .GetSection("AppSettings")
```

Comment:
Why appsettings section? Shouldn't you have a MailChimp section if you want a section for it?